scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/SimulationFacade.java

```diff
@@ -80,66 +89,65 @@ public PlanningHorizon getPlanningHorizon(){
-      if(lastSimDriverResults.unfinishedActivities.get(planActInstanceIdToSimulationActInstanceId.get(activityInstance.getId())) != null){
-        System.out.println("Activity "
-                           + activityInstance
-                           + " has not finished, check planning horizon ?");
-      } else{
-        System.out.println("Simulation has been launched but activity with name= "
-                           + activityInstance
-                           + " has not been found");
-      }
+    try {
+       var duration = driver.getActivityDuration(planActInstanceIdToSimulationActInstanceId.get(activityInstance.getId()));
+       if(duration.isPresent()){
+         return duration;
+       } else{
+         System.out.println("Incremental simulation is probably outdated, check that no activity is removed between simulation and querying");
+         return Optional.empty();
+       }
+    } catch(IllegalStateException exception){
+      System.out.println("Activity "
+                         + activityInstance
+                         + " has not finished, check planning horizon ?");
+      return Optional.empty();
     }
-    return null;
   }
 
-  /**
-   * Simulates schedule and processes results to produce accessors for resources and activities. The complete set of
-   * activities scheduled so
-   * far should be passed for every simulation. Simulation runs for the whole planning horizon.
-   *
-   * @param plan the plan to simulate
-   */
-  public void simulatePlan(Plan plan) {
-    planActInstanceIdToSimulationActInstanceId.clear();
-    final var actsInPlan = plan.getActivitiesByTime();
-    final var schedule = new HashMap<ActivityInstanceId, Pair<Duration, SerializedActivity>>();
-    long counter = 0;
-
-    for (final var act : actsInPlan) {
-
-      Map<String, SerializedValue> params = new HashMap<>();
-      params.putAll(act.getArguments());
-      if(act.getDuration()!= null) {
-        params.put("duration", new DurationValueMapper().serializeValue(act.getDuration()));
-      } else{
-        System.out.println("Warning : activity has no duration parameter");
+  public void removeActivitiesFromSimulation(Collection<ActivityInstance> activities){
+    var atLeastOne = false;
+    for(var act: activities){
+      if(insertedActivities.containsKey(act)){
+        atLeastOne = true;
+        insertedActivities.remove(act);
       }
-      var activityIdSim = new ActivityInstanceId(counter++);
-      planActInstanceIdToSimulationActInstanceId.put(act.getId(), activityIdSim);
-      schedule.put(activityIdSim, Pair.of(
-          act.getStartTime(),
-          new SerializedActivity(act.getType().getName(), params)));
     }
+    //reset incremental simulation
+    if(atLeastOne){
+      var oldInsertedActivities = new HashMap<>(insertedActivities);
+      insertedActivities.clear();
+      planActInstanceIdToSimulationActInstanceId.clear();
+      driver = new IncrementalSimulationDriver(missionModel);
```

Comment:
Just checking - is creating a new IncrementalSimulationDriver the current expected way to reset the simulation?

Reply:
The other way was to call `IncrementalSimulation.initSimulation` but it is currently private. Both solutions are very close in terms of what's done.